python/nano/src/bigdl/nano/deps/inference/model.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import List, Tuple
+from bigdl.nano.pytorch.lightning import LightningModuleFromTorch
+import inspect
+from torch.utils.data import DataLoader
+import torch
+
+
+class AcceleratedLightningModule(LightningModuleFromTorch):
+    def __init__(self, model):
+        super().__init__(model)
+        self.on_init_end()
+
+    def on_init_end(self, *args):
+        pass
+
+    def forward(self, *inputs):
+        inputs = self.on_forward_start(inputs)
+        outputs = self.forward_step(*inputs)
+        return self.on_forward_end(outputs)
+
+    def train(self, mode=True):
```

Comment:
what does `mode` mean?

Reply:
`model.eval()` actually calls `train(mode=False)`.  
So with modifications on `train`, we can change the behavior for both functions.

Reply:
This is simply an overwrite of the original `model.train`, so they should have the same signature.
ref: https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.train